projects/kit/components/input-files/input-files.component.ts

```diff
@@ -0,0 +1,329 @@
+
+// @dynamic
+@Component({
+    selector: 'tui-input-files',
+    templateUrl: './input-files.template.html',
+    styleUrls: ['./input-files.style.less'],
+    changeDetection: ChangeDetectionStrategy.OnPush,
+    providers: [
+        MODE_PROVIDER,
+        {
+            provide: TUI_FOCUSABLE_ITEM_ACCESSOR,
+            useExisting: forwardRef(() => TuiInputFilesComponent),
+        },
+    ],
+})
+export class TuiInputFilesComponent
+    extends AbstractTuiNullableControl<TuiFileLike | ReadonlyArray<TuiFileLike>>
+    implements TuiFocusableElementAccessor
+{
+    private dataTransfer: DataTransfer | null = null;
+
+    @Input()
+    @tuiDefaultProp()
+    link: PolymorpheusContent = '';
+
+    @Input()
+    @tuiDefaultProp()
+    label: PolymorpheusContent = '';
+
+    @Input()
+    @tuiDefaultProp()
+    accept = '';
+
+    @Input()
+    @tuiDefaultProp()
+    multiple = false;
+
+    @Input()
+    @tuiDefaultProp()
+    size: TuiSizeL = 'm';
+
+    @Input()
+    @tuiDefaultProp()
+    maxFileSize = DEFAULT_MAX_SIZE;
+
+    @Output()
+    onReject = new EventEmitter<ReadonlyArray<TuiFileLike> | TuiFileLike>();
+
+    @ViewChild('input')
+    readonly input?: ElementRef<HTMLInputElement>;
+
+    constructor(
+        @Optional()
+        @Self()
+        @Inject(NgControl)
+        control: NgControl | null,
+        @Inject(ChangeDetectorRef)
+        changeDetectorRef: ChangeDetectorRef,
+        @Inject(TUI_IS_MOBILE)
+        readonly isMobile: boolean,
+        @Inject(TUI_INPUT_FILE_TEXTS)
+        readonly inputFileTexts$: Observable<
+            Record<
+                | 'defaultLabelSingle'
+                | 'defaultLabelMultiple'
+                | 'defaultLinkSingle'
+                | 'defaultLinkMultiple'
+                | 'maxSizeRejectionReason'
+                | 'formatRejectionReason'
+                | 'drop'
+                | 'dropMultiple',
+                string
+            >
+        >,
+        @Inject(TUI_MODE) readonly mode$: Observable<TuiBrightness | null>,
+        @Inject(TUI_DIGITAL_INFORMATION_UNITS)
+        readonly units$: Observable<[string, string, string]>,
+    ) {
+        super(control, changeDetectorRef);
+    }
+
+    get nativeFocusableElement(): TuiNativeFocusableElement | null {
+        return this.input ? this.input.nativeElement : null;
+    }
+
+    get focused(): boolean {
+        return isNativeFocused(this.nativeFocusableElement);
+    }
+
+    get computedLink$(): Observable<PolymorpheusContent> {
+        return this.computeLink$(this.fileDragged, this.multiple, this.link);
+    }
+
+    get computedLabel$(): Observable<PolymorpheusContent> {
+        return this.computeLabel$(
+            this.isMobile,
+            this.fileDragged,
+            this.multiple,
+            this.label,
+        );
+    }
+
+    get fileDragged(): boolean {
+        return !!this.dataTransfer && this.dataTransfer.types.indexOf('Files') !== -1;
```

Comment:
nit:
```suggestion
        return !!this.dataTransfer && this.dataTransfer.types.includes('Files');
```

Reply:
Gotta be sure it's actual `Array` and not something Array-like that doesn't have `includes` method.